### PR TITLE
Adds support for pebbledb

### PR DESCRIPTION
**Pull request by faddat**
Pebbledb is a rocks-like kv store written in pure go.

It has performance and resource consumption benefits when compared to rocksdb. 

Likely, its ease of use, configurability, and performance will make it the logical candidate for default when compared to rocksdb and goleveldb.

Errors need to be corrected before marking ready for review.

### CODE REVIEW

File: pkgs/db/pebble.go [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+			l.FilterPolicy = bloom.FilterPolicy(10)
+			l.FilterType = pebble.TableFilter
+			if i > 0 {
+				l.TargetFileSize = opts.Levels[i-1].TargetFileSize * 2
+			}
+			l.EnsureDefaults()
+		}
+	*/
+	//	opts.Levels[6].FilterPolicy = nil
+	//	opts.FlushSplitBytes = opts.Levels[0].TargetFileSize
```
Comment (moul) [inline pkgs/db/pebble.go:64]: Can you propose a first set of settings + a link to the official documentation so we can improve it later?

**Review by moul — COMMENTED** [review]
I also think that Pebbledb should become the new default.

And maybe not only the « default », but the only available option to make things even more simple, to reduce dependencies and inconsistencies.

**Comment by ajnavarro** [comment]
Hi @faddat !

Can we help you with something to move this draft forward? Thanks!

**Comment by moul** [comment]
This topic is currently being managed by @jaekwon on tm2. We can close it for now, and if it becomes relevant again in the future, he can revive the discussion. Thank you.

**Pull request closed without merging**